host/metrics.go

```diff
@@ -0,0 +1,215 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+	"sync"
+
+	host "github.com/flynn/flynn/host/types"
+	"github.com/inconshreveable/log15"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func registerMetrics(state *State, log log15.Logger) error {
+	labels := prometheus.Labels{
+		"host": state.id,
+	}
+
+	return prometheus.Register(&metrics{
+		state: state,
+		log:   log,
+		memFreeDesc: prometheus.NewDesc(
+			"host_memory_total_bytes",
+			"Host total memory in bytes",
+			nil, labels,
+		),
+		memAvailDesc: prometheus.NewDesc(
+			"host_memory_avail_bytes",
+			"Host available memory in bytes",
+			nil, labels,
+		),
+		load1mDesc: prometheus.NewDesc(
+			"host_load_1m",
+			"Host 1m load average",
+			nil, labels,
+		),
+		load5mDesc: prometheus.NewDesc(
+			"host_load_5m",
+			"Host 5m load average",
+			nil, labels,
+		),
+		load15mDesc: prometheus.NewDesc(
+			"host_load_15m",
+			"Host 15m load average",
+			nil, labels,
+		),
+		activeJobsDesc: prometheus.NewDesc(
+			"host_jobs_active",
+			"Host active jobs",
+			nil, labels,
+		),
+		startingJobsDesc: prometheus.NewDesc(
+			"host_jobs_starting",
+			"Host starting jobs",
+			nil, labels,
+		),
+		runningJobsDesc: prometheus.NewDesc(
+			"host_jobs_running",
+			"Host running jobs",
+			nil, labels,
+		),
+	})
+}
+
+type metrics struct {
+	state *State
+	log   log15.Logger
+
+	memFreeDesc  *prometheus.Desc
+	memAvailDesc *prometheus.Desc
+
+	load1mDesc  *prometheus.Desc
+	load5mDesc  *prometheus.Desc
+	load15mDesc *prometheus.Desc
```

Comment:
Would it make more sense to add the [node exporter](https://github.com/prometheus/node_exporter)?

Reply:
I did initially try and use it but I hit a couple of issues:

* there were vendoring issues due to our outdated docker fork (which will be resolved by #4367)
* it is tied quite tightly with [a flag parsing library](https://github.com/prometheus/node_exporter/blob/master/collector/collector.go#L71), so calling [`NewNodeCollector`](https://github.com/prometheus/node_exporter/blob/master/collector/collector.go#L83) doesn't actually give us any metrics (they're all disabled)

I decided to just do a few metrics manually until we get round to being able to use [cAdvisor](https://github.com/google/cadvisor).

Reply:
The standard output of cAdvisor with small modification (join the name and type with the container id) provides great insights and might be worth aiming for in the first release (at least for container cpu and memory).

Recently configured it on my instance with a small replace script between cAdvisor and Prometheus
![flynn-metrics](https://user-images.githubusercontent.com/654630/39422775-50d6be94-4c5e-11e8-9dc9-3e44f2ec6fce.png)

Reply:
@BrynjarM nice! We'll definitely be integrating cAdvisor, there are just some blockers before we can do so that I don't want to get in the way of merging this PR.